Add tests

itgavrilov opened the pull request.

itgavrilov committed:
- рефакторинг
- class IcdTest

itgavrilov committed:
- рефакторинг
- поправил глюк со стартом записи(пробелма была в добавлении элементов в другом патоке)
- class ChannelTest
- class DeviceTest

itgavrilov committed:
- рефакторинг
- class PatientRecordTest
- class ExaminationTest

itgavrilov committed:
- !!!перестроил архитектуру!!!

Stanru commented on src/main/java/ru/gsa/biointerface/services/ServiceDevice.java:
```diff
@@ -0,0 +1,105 @@
+/**
+ * Created by Gavrilov Stepan (itgavrilov@gmail.com) on 10.09.2021.
+ */
+public class ServiceDevice {
```
названия наоборот DeviceService

Stanru commented on src/main/java/ru/gsa/biointerface/services/ServiceException.java:
```diff
@@ -0,0 +1,22 @@
+/**
+ * Created by Gavrilov Stepan (itgavrilov@gmail.com) on 10.09.2021.
+ */
+public class ServiceException extends Exception {
```
Все ошибки в отдельный пакет exception

Stanru replied:
хотя можно и так оставить

Stanru commented on src/main/java/ru/gsa/biointerface/ui/window/ChannelAddController.java:
```diff
@@ -24,7 +26,12 @@ public void showWindow() throws UIException {
 
-        transitionGUI.show();
+        try {
+            serviceChannel = ServiceChannel.getInstance();
```
Лучше в конструктор вынести

Stanru commented on src/main/java/ru/gsa/biointerface/ui/window/ChannelAddController.java:
```diff
@@ -4,14 +4,16 @@
  * Created by Gavrilov Stepan (itgavrilov@gmail.com) on 10.09.2021.
  */
 public class ChannelAddController extends AbstractWindow {
+    private ServiceChannel serviceChannel;
```
Сделай везде сервисы final, заставят тебя инициализацию при создании объекта сделать)

Stanru commented:
`String str = channelField.getText().trim().replaceAll("\s.*", "").replaceAll("[^a-zA-Zа-яА-Я0-9.:]", "");`
такие названия str не пойдут. Строка более 120 символов

Stanru commented:
`generateNewWindow("fxml/Channels.fxml").showWindow();`
В классе в статик финал

itgavrilov committed:
- ренейминг
- рефакторинг
- переосмысление работы с исключениями
- расширение логирования

itgavrilov committed:
- добавил в UI окно сообщения об ошибке

itgavrilov committed:
- рефакторинг
- PatientRecordServiceTest
- IcdServiceTest
- ExaminationServiceTest
- DeviceServiceTest
- ChannelNameServiceTest

itgavrilov committed:
- немного почистил
